schemas/v3.1/meta/oas.json

```diff
@@ -0,0 +1,1424 @@
+{
+  "$schema": "http://json-schema.org/draft/2019-WIP/schema#",
+  "$id": "https://spec.openapis.org/oas/3.1/meta/oas",
+  "$vocabulary": {
+    "https://specs.openapis.org/oas/3.1/vocab/oas": true
+  },
+  "$recursiveAnchor": true,
+  "title": "OpenAPI Schema Object Vocabulary Schema",
+  "type": ["object", "boolean"],
+  "properties": {
+    "nullable": {
+      "type": "boolean",
+      "default": false,
+      "deprecated": true
+    },
+    "discriminator": {
+      "$ref": "#/$defs/Discriminator"
+    },
+    "example": {
+      "deprecated": true
+    },
+    "externalDocs": {
+      "$ref": "#/$defs/ExternalDocumentation"
+    },
+    "xml": {
+      "$ref": "#/$defs/XML"
+    }
+  },
+  "patternProperties": {
+    "^x-": {
+    }
+  },
+  "unevaluatedProperties": false,
+  "$defs": {
```

Comment:
While it make sense to have `Discriminator` and `XML`, which are used in this file, defined here, I'm a little unclear on why you put all of the other definitions here.  They are describing other aspects of the OAS format, not parts of the schema, so they don't make sense to me for storing in the vocabulary meta-schema.  I'd put them in `schemas/v3.1/schema.json`

Since OAS is a document format that embeds schemas, the thing that describes the schema format is a meta-schema.  The thing that describes the overall format is not a meta-schema, it's just a schema.  Schemas have a root schema at the.. um... root of the document, which is not how the root object of OAS behaves.